File: src/main/java/io/github/asyncronous/exo/block/BlockTest.java
package io.github.asyncronous.exo.block;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;

import io.github.asyncronous.exo.Cores;
import io.github.asyncronous.exo.EXOTag;
import io.github.asyncronous.exo.Exoskeleton;
import io.github.asyncronous.exo.item.ItemExoskeletonArmor;

public class BlockTest
extends Block{
    public BlockTest(){
        super(Material.iron);
        this.setCreativeTab(Exoskeleton.tab);
    }

    @Override
    public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int p_149727_6_, float p_149727_7_, float p_149727_8_, float p_149727_9_){
        ItemStack stack = player.getCurrentEquippedItem();
        if(stack != null){
            if(stack.getItem() instanceof ItemExoskeletonArmor){
                ItemExoskeletonArmor armor = (ItemExoskeletonArmor) stack.getItem();
                Cores.coreReflex.write(EXOTag.getTag(stack), armor);
            }
        }

        return true;
    }
}

File: src/main/scala/exoskeleton/common/block/BlockTest.scala
package exoskeleton.common.block

import exoskeleton.common.Exoskeleton
import exoskeleton.common.item.ItemExoArmor
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.item.ItemStack
import net.minecraft.world.World

object BlockTest
extends Block(Material.iron) {
  this.setCreativeTab(Exoskeleton.tab);
  this.setBlockName("debug");

  override def onBlockActivated(world: World, x: Int, y: Int, z: Int, player: EntityPlayer, p_149727_6_ : Int, p_149727_7_ : Float, p_149727_8_ : Float, p_149727_9_ : Float): Boolean = {
    val stack: ItemStack = player.getCurrentEquippedItem();

    if(stack != null &&
       stack.getItem().isInstanceOf[ItemExoArmor]) {
      player.openGui(Exoskeleton, 3, world, x, y, z);
    }

    return true;
  }
}